Controllers/MoviesController.cs

```diff
@@ -0,0 +1,277 @@
+                var movies = _service.GetMovies().Select(m => new MovieDTO
+                {
+                    Id = m.Id,
+                    Name = m.Name,
+                    Director = m.Director,
+                    AverageRating = _reviewService.GetAverageRating(m.Id),
+                    Studios = m.Studios.Select(s => new StudioSimplified { Id = s.Id, Name = s.Name }).ToList(),
+                    Year = m.Year,
+                    Description = m.Description,
+                    Genre = m.Genre
+                });
+                return Ok(movies.Where(m => m.Genre.ToLower() == genre.ToLower()));
+            } catch
+            {
+                return NotFound($"No movies associated with the {genre} genre.");
+            }
+
+        } 
+
+        [HttpGet("{id}")]
+        public IActionResult Get(int id)
+        {
+            try
+            {
+                var movie = _service.GetMovieById(id);
+                return Ok(new MovieDTO
+                {
+                    Id = movie.Id,
+                    Name = movie.Name,
+                    Director = movie.Director,
+                    Studios = movie.Studios.Select(s => new StudioSimplified { Id = s.Id, Name = s.Name }).ToList(),
+                    Year = movie.Year,
+                    Description = movie.Description,
+                    Genre = movie.Genre
+                });
+            } catch
+            {
+                return NotFound($"Movie with ID {id} not found.");
+            }
+        }
+
+        [HttpPost]
+        public IActionResult Post([FromBody] CreateMovieDTO movieDTO)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            try
+            {
+                var movie = new Movie
+                {
+                    Name = movieDTO.Name,
+                    Year = movieDTO.Year,
+                    Description = movieDTO.Description,
+                    Director = movieDTO.Director,
+                    Genre = movieDTO.Genre
+                };
+                _service.AddMovie(movie);
+
+                var responseMovie = new MovieDTO
+                {
+                    Id = movie.Id,
+                    Name = movie.Name,
+                    Year = movie.Year,
+                    Description = movie.Description,
+                    Genre = movie.Genre,
+                    Director = movie.Director
+                };
+
+                return CreatedAtAction(nameof(Get), new { id = movie.Id }, responseMovie);
+            } catch
+            {
+                return BadRequest("Encountered an error when creating the movie.");
+            }
+        }
+
+        [Route("/api/Movies/AddStudio")]
+        [HttpPost]
+        public IActionResult Post([FromBody] MovieStudio movieStudio)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
```

Comment:
create a validator which checks the model state, this way you will avoid a lot of code duplication